packages/terra-date-time-picker/src/DateTimePicker.jsx

```diff
@@ -0,0 +1,361 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import 'terra-base/lib/baseStyles';
+import DatePicker from 'terra-date-picker';
+import TimeInput from 'terra-time-input';
+import DateUtil from 'terra-date-picker/lib/DateUtil';
+import styles from './DateTimePicker.scss';
+import DateTimeUtils from './DateTimeUtils';
+import TimeClarification from './_TimeClarification';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * An array of ISO 8601 string representation of the dates to disable in the picker.
+   */
+  excludeDates: PropTypes.arrayOf(PropTypes.string),
+  /**
+   * A function that gets called for each date in the picker to evaluate which date should be disabled.
+   * A return value of true will be enabled and false will be disabled.
+   */
+  filterDate: PropTypes.func,
+  /**
+   * An array of ISO 8601 string representation of the dates to enable in the picker.
+   * All Other dates will be disabled.
+   */
+  includeDates: PropTypes.arrayOf(PropTypes.string),
+  /**
+   * Custom input attributes to apply to the date and time inputs. Use the name prop to set the name for the input.
+   * Do not set the name in inputAttribute as it will be ignored.
+   */
+  inputAttributes: PropTypes.object,
+  /**
+   * An ISO 8601 string representation of the maximum date time.
+   */
+  maxDateTime: PropTypes.string,
+  /**
+   * An ISO 8601 string representation of the minimum date time.
+   */
+  minDateTime: PropTypes.string,
+  /**
+   * Name of the date input. The name should be unique.
+   */
+  name: PropTypes.string.isRequired,
+  /**
+   * A callback function to execute when a valid date is selected or entered.
+   * The first parameter is the event. The second parameter is the changed input value.
+   */
+  onChange: PropTypes.func,
```

Comment:
Seems like the parameters should be flipped. From most if not all consumers perspective, I could care less what event triggered the change. I just want to know that the value changed so I can store the value off. If the parameters are flipped then really only those that care about the event have to add them to their handler.

Reply:
I agree with your suggestion and don't mind making the flip. However, callback props in our other React components have the event as the first parameter and additional data as the second parameter. Flipping them here would make it inconsistent with other React components.

Reply:
I would argue those components are following a bad pattern